contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/MapRDBPushFilterIntoScan.java

```diff
@@ -137,11 +137,13 @@ protected void doPushFilterIntoJsonGroupScan(RelOptRuleCall call,
       return; //no filter pushdown ==> No transformation.
     }
 
+    // Pass tableStats from old groupScan so we do not go and fetch stats (an expensive operation) again from MapR DB client.
     final JsonTableGroupScan newGroupsScan = new JsonTableGroupScan(groupScan.getUserName(),
                                                                     groupScan.getStoragePlugin(),
                                                                     groupScan.getFormatPlugin(),
                                                                     newScanSpec,
-                                                                    groupScan.getColumns());
+                                                                    groupScan.getColumns(),
+                                                                    groupScan.getTableStats());
```

Comment:
We should try to use `clone()` here. All we are doing is copying stuff from one groupscan to another. `JsonTableGroupScan` already has a clone which clones everything except columns.

`@Override 
public GroupScan clone(List<SchemaPath> columns) 
{ JsonTableGroupScan newScan = new JsonTableGroupScan(this); 
newScan.columns = columns; return newScan; }`
We can create another which would clone everything except scanSpec. This can be used to pass in the `newScanSpec` generated here. Doing this would also clone the `regionsToScan` saving the call to `init()`.